src/allocator/mod.rs

```diff
@@ -0,0 +1,197 @@
+//! Defines [`Allocator`] interface used to bound allocation of buffers
+//! for user data transmission inside NFS-Mamont implementation.
+
+mod slice;
+
+use std::num::NonZeroUsize;
+
+use async_trait::async_trait;
+use tokio::sync::mpsc;
+
+pub use slice::Slice;
+
+type Sender<T> = mpsc::UnboundedSender<T>;
+type Receiver<T> = mpsc::UnboundedReceiver<T>;
+type Buffer = Box<[u8]>;
+
+/// Allocates [`Slice`]'s.
+#[async_trait]
+pub trait Allocator {
+    /// Returns [`Slice`] of specified size.
+    ///
+    /// # Parameters
+    ///
+    /// - `size` --- size of returned slice.
+    ///
+    /// # Panic
+    ///
+    /// This method panics if size is greater then allocator capacity.
+    async fn alloc(&mut self, size: NonZeroUsize) -> Option<slice::Slice>;
+}
+
+pub struct Impl {
```

Comment:
I'm not quite sure, but probably we should use names like RecyleAllocator or CyclicAllocator

Reply:
Not to contain work allocator in public entity in module `allocator`. We should use modules for such a naming.

Ok, we throw away postifx at that point, then `Recycle` or `Cyclic` allocator remains. Every allocator is Recycle and Cyclic since it allocates and then deallocates. Why not just allocator::Impl? It will be invisible to users after allocator::Service implementation